src/mds/Server.cc

```diff
@@ -2365,7 +2364,7 @@ void Server::handle_client_request(const cref_t<MClientRequest> &req)
   bool sessionclosed_isok = replay_unsafe_with_closed_session;
   // active session?
   Session *session = 0;
-  if (req->get_source().is_client()) {
+  if (req->get_connection()->get_peer_type() == CEPH_ENTITY_TYPE_CLIENT) {
```

Comment:
This is a common pattern. Add a helper method to `src/msg/Message.h`.

Reply:
okay

Reply:
Added the helper method.